test/e2e/network/scale/ingress.go

```diff
@@ -0,0 +1,472 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scale
+
+import (
+	"fmt"
+	"io/ioutil"
+	"sync"
+	"time"
+
+	"k8s.io/api/core/v1"
+	extensions "k8s.io/api/extensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	clientset "k8s.io/client-go/kubernetes"
+
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+const (
+	numIngressesSmall  = 5
+	numIngressesMedium = 20
+	numIngressesLarge  = 100
+
+	scaleTestIngressNamePrefix = "ing-scale"
+	scaleTestBackendName       = "echoheaders-scale"
+	scaleTestSecretName        = "tls-secret-scale"
+	scaleTestHostname          = "scale.ingress.com"
+	scaleTestNumBackends       = 10
+	scaleTestPollInterval      = 15 * time.Second
+
+	maxConcurrentIngressCreations = 20
+	// We don't expect waitForIngress to take longer
+	// than waitForIngressMaxTimeout.
+	waitForIngressMaxTimeout = 60 * time.Minute
+	ingressesCleanupTimeout  = 60 * time.Minute
+)
+
+var (
+	scaleTestLabels = map[string]string{
+		"app": scaleTestBackendName,
+	}
+)
+
+// Limit the number of parallel ingress creations.
+var threadLimit = make(chan struct{}, maxConcurrentIngressCreations)
```

Comment:
Now I think this might be unnecessary. Doesn't make sense to throttle ingress creations.

Reply:
Removed.